### PR TITLE
Some optimizations

### PR DESCRIPTION
**Pull request by archermarx**
# What's changed
- made version of `interpolate` specialized for `StepRangeLen`
- Swapped out `Tev^1.5` in electron-ion collision frequency computation for `sqrt(Tev^3)`
- Compute nu_iz and ionization component of inelastic losses during heavy species update
- remove compute_z_eff and rearrange some computations in update_electrons

Using an example SPT-100 simulation with 150 cells and only singly-charged ions as a benchmark, I recorded a time of around 1.56 seconds before these optimizations and 1.00 seconds after these optimizations. That's about a 36% speedup, due primarily to the first change and third changes. For simulations with more charge states, these gains will be larger as the primary effect of the `interpolate` and ionization rate changes is to speed up reaction rate calculation.

Profiling shows that reaction rate computation is still a major bottleneck for the code. Next PR will focus on improving this by separating out reaction computation into its own step in the main solver loop.

### CODE REVIEW

File: src/collisions/collision_frequencies.jl [hunk 1]
```diff
@@ -20,42 +20,13 @@ end
     freq_electron_ion(ne, Tev, Z)
 Effective frequency at which electrons are scattered due to collisions with ions
 """
-@inline freq_electron_ion(ne::Number, Tev::Number, Z::Number) = 2.9e-12 * Z^2 * ne * coulomb_logarithm(ne, Tev, Z) / Tev^1.5
-
-function compute_Z_eff(U, params, i::Int)
```
Comment (DecBrick) [inline src/collisions/collision_frequencies.jl:25]: The Braginskii closure uses the effective charge state to compute the coefficient value, so this function may still be needed

File: src/simulation/sourceterms.jl [hunk 2]
```diff
@@ -42,6 +45,7 @@ function apply_reactions!(dU::AbstractArray{T}, U::AbstractArray{T}, params, i::
         end
     end
 
+   # @show νiz[i]
```
Comment (DecBrick) [inline src/simulation/sourceterms.jl:48]: Vestigial comment

File: src/collisions/collision_frequencies.jl (hunk 1 above)
Comment (archermarx) [reply src/collisions/collision_frequencies.jl:25]: We still compute it in the update_electrons function, but we do it at the same time as other ion quantities for speed.

**Comment by archermarx** [comment]
I still need to fix tests, one moment

File: test/order_verification/ovs_energy.jl [hunk 3]
```diff
@@ -35,7 +35,7 @@ ue_func = eval(build_function(expand_derivatives(ue), [x]))
 ρn_func = eval(build_function(ρn, [x]))
 ϵ_func = eval(build_function(ϵ, [x]))
 
-k(ϵ) = 12.12 * OVS_rate_coeff_iz(ϵ) + 8.32 * OVS_rate_coeff_ex(ϵ)
+k(ϵ) = 8.32 * OVS_rate_coeff_ex(ϵ)
```
Comment (DecBrick) [inline test/order_verification/ovs_energy.jl:38]: Seems like the ionization losses were removed from the test. Are they included elsewhere? 

Also might be worth adding a comment to note that the constants here are for Xenon

**Pull request merged**